CMakeLists.txt

```diff
@@ -15,9 +15,42 @@ set_target_properties(ArborX PROPERTIES INTERFACE_COMPILE_FEATURES cxx_std_14)
 # within a subdirectory.
 add_dependencies(ArborX record_hash)
 
+function(set_amd_gpu_target)
+  if(NOT DEFINED Kokkos_ENABLE_HIP)
+    return()
+  endif()
+
+  if(VEGA900 IN_LIST Kokkos_ARCH)
+    set(Kokkos_AMD_GPU_TARGET gfx900)
+  elseif(VEGA906 IN_LIST Kokkos_ARCH)
+    set(Kokkos_AMD_GPU_TARGET gfx906)
+  elseif(VEGA908 IN_LIST Kokkos_ARCH)
+    set(Kokkos_AMD_GPU_TARGET gfx908)
+  elseif(VEGA90A IN_LIST Kokkos_ARCH)
+    set(Kokkos_AMD_GPU_TARGET gfx90a)
+  else()
+    message(WARNING "GPU device code will be built for all architectures suported by ROCm.")
+    return()
+  endif()
+
+  if(NOT DEFINED GPU_TARGETS)
+    set(GPU_TARGETS ${Kokkos_AMD_GPU_TARGET} PARENT_SCOPE)
+    return()
+  endif()
+
+  list(LENGTH GPU_TARGETS NUM_AMD_GPU_ARCHITECTURES)
+  if(NUM_AMD_GPU_ARCHITECTURES GREATER 1)
+    message(WARNING "More than one AMD GPU architecture has been provided.\nGPU_TARGETS: ${GPU_TARGETS}")
+  elseif(NOT GPU_TARGETS STREQUAL Kokkos_AMD_GPU_TARGET)
+    message(WARNING "Provided AMD GPU target does not match the architecture enabled in Kokkos (${Kokkos_AMD_GPU_TARGET})\nGPU_TARGETS: ${GPU_TARGETS}")
+  endif()
+endfunction()
+
 include(CMakeDependentOption)
 cmake_dependent_option(ARBORX_ENABLE_ROCTHRUST "Enable rocThrust support" ON "Kokkos_ENABLE_HIP" OFF)
 if(Kokkos_ENABLE_HIP AND ARBORX_ENABLE_ROCTHRUST)
+  set_amd_gpu_target()
```

Comment:
@Rombur let me know what you think about the function.
We would need to put in a separate CMake file that gets installed and call it again in our exported configuration before finding the rocThrust dependency.

Reply:
ping @Rombur